ferry.go

```diff
@@ -890,6 +908,21 @@ func (f *Ferry) ReportProgress() {
 	}
 }
 
+func (f *Ferry) ReportState() {
+	callback := f.Config.StateCallback
+	state, err := f.SerializeStateToJSON()
+	if err != nil {
+		f.logger.WithError(err).Error("failed to dump state to JSON")
+		return
```

Comment:
Should this panic here via the ErrorHandler?

Reply:
Hmm, I think no but don't feel strongly and could be convinced otherwise. 

This would _not_ prevent the run from completing, and worst case the run would need to restart from the beginning. 

What do you think?

Reply:
Failing to dump the JSON seems like a weird condition that we would probably panic?